### PR TITLE
Remote attestation

### PR DESCRIPTION
**Pull request by DenisKarch**
Added functionality that is commonly used for remote attestation.

Added:
  - GetCapability
  - GetManufacturer
  - NVReadValue
  - ReadEKCert
  - ActivateIdentity

### CODE REVIEW

File: tpmutil/structures.go [hunk 1]
```diff
@@ -48,3 +48,6 @@ type responseHeader struct {
 
 // A Handle is a reference to a TPM object.
 type Handle uint32
+
+// A U16Bytes is a byte slice with 16-bit header.
+type U16Bytes []byte
```
Comment (ericchiang) [inline tpmutil/structures.go]: Why is this necessary? The encoding logic looks identical to []byte.

File: tpm/tpm.go [hunk 2]
```diff
@@ -713,6 +851,103 @@ func OwnerReadSRK(rw io.ReadWriter, ownerAuth digest) ([]byte, error) {
 	return tpmutil.Pack(pk)
 }
 
+// ReadEKCert reads the EKCert from the NVRAM
+func ReadEKCert(rw io.ReadWriter, ownAuth digest) ([]byte, error) {
+	var (
```
Comment (ericchiang) [inline tpm/tpm.go]: nit: remove this var block. you don't need it and can just use `:=` to allocate new variables.

File: tpm/tpm.go [hunk 3]
```diff
@@ -713,6 +851,103 @@ func OwnerReadSRK(rw io.ReadWriter, ownerAuth digest) ([]byte, error) {
...
+		ekbuf    []byte
+		buff     io.Reader
+		offset   uint32
+		bufSize  uint32
+		tag      uint16
+		certType uint8
+		certSize uint16
+	)
+	offset = 0
+	data, err := NVReadValue(rw, 0x1000f000, offset, 5, ownAuth)
```
Comment (ericchiang) [inline tpm/tpm.go]: if you ever have a raw value like this do the following instead:

```
const descriptiveName = 0x1000f000
data, err := NVReadValue(rw, descriptiveName, offset, 5, ownAuth)
```

File: tpmutil/structures.go (hunk 1 above)
Comment (DenisKarch) [reply tpmutil/structures.go]: The encoding is identical for []byte on TPM2.0. However tpmutil assumes that TPM1.2 uses a 32-bit header which--although true in most cases--is not the case for the TPM_SYMMETRIC_KEY that is returned from ActivateIdentity. I mentioned this in the commit comment but I'll add a comment in the file as well.

**Comment by DenisKarch** [comment]
PR split into 3
getCapability
https://github.com/google/go-tpm/pull/82
ReadEKCert
https://github.com/google/go-tpm/pull/83
ActivateIdentity
https://github.com/google/go-tpm/pull/84
